dll/win32/shell32/wine/changenotify.c

```diff
@@ -220,6 +617,67 @@ SHChangeNotifyRegister(
     int cItems,
     SHChangeNotifyEntry *lpItems)
 {
+#ifdef __REACTOS__
+    HANDLE hMutex;
+    DWORD pid;
+    UINT nRegID = INVALID_REGID;
+    DWORD cbBlock;
+    LPBLOCK lpBlock;
+    INT iItem;
+    LPITEM pItem;
+
+    TRACE("(%p,0x%08x,0x%08x,0x%08x,%d,%p)\n",
+          hwnd, fSources, wEventMask, uMsg, cItems, lpItems);
+
+    if (cItems <= 0 || !lpItems || !IsWindow(hwnd))
+    {
+        ERR("Sanity check\n");
+        return nRegID;
+    }
+
+    GetWindowThreadProcessId(hwnd, &pid);
+    if (pid != GetCurrentProcessId())
+    {
+        ERR("Sanity check 2\n");
+        return nRegID;
+    }
+
+    hMutex = CreateMutexW(NULL, TRUE, CHANGENOTIFY_MUTEX_NAME);
```

Comment:
This does not offer any protection. If the mutex does not yet exist, this will create and acquire it. However, if it already exists (i.e. another thread/process is holding it), this will open the mutex without acquiring it. So in that case, the `ReleaseMutex` call below will fail because the mutex is not owned... and we won't actually get mutual exclusion.

One way to fix this would be to set bInitialOwner to FALSE and always call `WaitForSingleObject`. Another would be to call `WaitForSingleObject` in case `CreateMutex` sets the last error to ERROR_ALREADY_EXISTS or whatever.

Checking for failure would also be nice.

Reply:
OK.